### PR TITLE
Add affinity groups to y-flow feature

### PR DESCRIPTION
**Pull request by dpoltavets**
**MERGE NOTE:**  merge after #4526 and change the base correspondingly.

### CODE REVIEW

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/yflow/update/action/ValidateYFlowAction.java [hunk 1]
```diff
@@ -102,6 +102,9 @@ public ValidateYFlowAction(PersistenceManager persistenceManager, FlowOperations
                     format("Provided sub-flow ids is not valid for the y-flow %s", yFlowId));
         }
 
+        String mainAffinityFlowId = yFlow.getSubFlows().iterator().next().getFlow().getAffinityGroupId();
```
Comment (sergii-iakovenko) [inline src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/yflow/update/action/ValidateYFlowAction.java:105]: I'd replace "yFlow.getSubFlows().iterator().next()" with yFlow.getSubFlows().stream().findFirst().orElseThrow(...) to validate the y-flow consistency.

Comment (dpoltavets) [reply src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/yflow/update/action/ValidateYFlowAction.java]: Fixed at your suggestion.

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/yflow/create/action/CreateSubFlowsAction.java [hunk 2]
```diff
@@ -42,11 +43,13 @@ public void perform(State from, State to, Event event, YFlowCreateContext contex
         String yFlowId = stateMachine.getYFlowId();
         Collection<RequestedFlow> requestedFlows =
                 YFlowRequestMapper.INSTANCE.toRequestedFlows(stateMachine.getTargetFlow());
+        stateMachine.setRequestedFlows(requestedFlows);
         log.debug("Start creating {} sub-flows for y-flow {}", requestedFlows.size(), yFlowId);
         stateMachine.clearCreatingSubFlows();
 
-        requestedFlows.forEach(requestedFlow -> {
+        Optional.ofNullable(requestedFlows.iterator().next()).ifPresent(requestedFlow -> {
```
Comment (sergii-iakovenko) [inline src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/yflow/create/action/CreateSubFlowsAction.java:50]: Calling next() without checking for hasNext() is the way to get NoSuchElementException, not a null value.

Comment (dpoltavets) [reply src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/yflow/create/action/CreateSubFlowsAction.java]: Now `.stream().findFirst()` is used.

**Review by sergii-iakovenko — APPROVED** [review]

**Review by niksv — APPROVED** [review]

**Pull request merged**